lib/AST/ModuleLoader.cpp

```diff
@@ -171,6 +171,19 @@ void ModuleLoader::findOverlayFiles(SourceLoc diagLoc, ModuleDecl *module,
   using namespace llvm::sys;
   using namespace file_types;
 
+  // If cross import information is passed on command-line, only use that.
+  auto &crossImports = module->getASTContext().SearchPathOpts.CrossImportInfo;
+  if (!crossImports.empty()) {
```

Comment:
Could it cause an issue if there are no cross-module imports but we try to search for them anyway in include-tree filesystem, or does it just exit via `if (file->getModuleDefiningPath().empty()) return;`?

Reply:
We bypass file system search if the cross module info is passed on command-line. If `crossImports` is not empty, we always return immediately no matter if we found overlay file or not.

Reply:
But what if there are no cross module imports on the command-line? Will we still search?

Reply:
Good call. Maybe we will add another flag to say not searching file system and pass that from scanner?